### PR TITLE
Jasmine: add missing MAX_PRETTY_PRINT_DEPTH property

### PR DESCRIPTION
**Pull request by smasala**
Please fill in this template.

- [x] Make your PR against the `master` branch.
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] _**(forked version has errors in other packages)**_ Run `tsc` without errors.
- [x] _**(forked version has lints - not related to these changes)**_ Run `npm run lint package-name` if a `tslint.json` is present.

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<https://github.com/jasmine/jasmine/issues/1291#issuecomment-286818531>>
- [x] _(not needed)_ Increase the version number in the header if appropriate.
- [x] _(not needed)_ If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "../tslint.json" }`.

### CODE REVIEW

**Comment by yuit** [comment]
@smasala the CI is broken regarding `Matcher` doesn't have identical type parameters. It doesn
t seem related to your change...could you try merge master and re run?

**Comment by smasala** [comment]
@yuit I merged and re-ran - still no cigar.

**Comment by paulvanbrenk** [comment]
Looks like when the change to `Matchers` was made in the main jasmine definition this got broken, can you give fixing that a try? I think just adding type parameters should resolve the issue.

**Comment by andy-ms** [comment]
We've recently updated the repository structure.
All packages are now in a `types` subdirectory.
Please rebase this pull request onto the latest `master` branch.

**Comment by smasala** [comment]
@paulvanbrenk will have a look

**Comment by smasala** [comment]
How can you test just certain packages? `npm test` ran through all the projects locally which took ages 👎

**Comment by andy-ms** [comment]
Just use `tsc` (in `types/jasmine`) and `npm run lint jasmine`. The full test runner will rarely find anything that those two don't.

File: types/protractor-helpers/index.d.ts [hunk 1]
```diff
@@ -44,7 +44,7 @@ declare namespace protractor {
 //        jasmine d.ts.
 
 declare namespace jasmine {
-  interface Matchers {
+  interface Matchers<T> {
```
Comment (andy-ms) [inline types/protractor-helpers/index.d.ts]: How is this related to the other changes?

Comment (smasala) [inline types/protractor-helpers/index.d.ts]: request from @paulvanbrenk (see above) so that travis would build the original file changes successfully.

**Comment by smasala** [comment]
strange about the local tests. I was unable to replicate the travis build errors locally unless I ran the test on all packages.
